packages/voting-contract/src/lib.rs

```diff
@@ -157,10 +162,11 @@ where
     P: Serialize + DeserializeOwned,
 {
     let mut proposal = proposals::<P>().load(storage, proposal_id)?;
-
+    // Update Status
+    proposal.update_status(&env.block);
     // We allow execution even after the proposal "expiration" as long as all votes come in before
     // that point. If it was approved on time, it can be executed any time.
-    if proposal.current_status(&env.block) != Status::Passed {
+    if proposal.status != Status::Passed {
```

Comment:
This is redundant, since previous call `current_status` is called by `update_status` anyway.

Reply:
I reverted back 169 to the way it was but curious would it not be more redundant to to keep it that way as `proposal.status` uses the result of the first `current_status` call rather than what I went back to which is calling it again?